queries/schedules.py

```diff
@@ -90,271 +95,831 @@ def help_text(lemma: str) -> str:
+    result["qdate"] = datetime.date.today() + datetime.timedelta(days=1)
 
-_HANDLER_MAP = {
-    _TELEVISION_QKEY: _gen_curr_tv_program_answer,
-    _TELEVISION_EVENING_QKEY: _gen_evening_tv_program_answer,
-    _RADIO_QKEY: _gen_curr_radio_program_answer,
+
+def QSchTomorrowEvening(node: Node, params: ParamList, result: Result) -> None:
+    result["qdate"] = datetime.date.today() + datetime.timedelta(days=1)
+    result["qtime"] = datetime.time(20, 0)
+    result["PM"] = True
+
+
+def QSchYesterday(node: Node, params: ParamList, result: Result) -> None:
+    result["qdate"] = datetime.date.today() - datetime.timedelta(days=1)
+
+
+def QSchYesterdayEvening(node: Node, params: ParamList, result: Result) -> None:
+    result["qdate"] = datetime.date.today() - datetime.timedelta(days=1)
+    result["qtime"] = datetime.time(20, 0)
+    result["PM"] = True
+
+
+def QSchNow(node: Node, params: ParamList, result: Result) -> None:
+    now = datetime.datetime.today()
+    result["qdate"] = now.date()
+    result["qtime"] = now.time()
+
+
+_STATION_ENDPOINTS = {
+    "Stöð 2": "https://api.stod2.is/dagskra/api/{0}/{1}",
+    "RÚV": "https://muninn.ruv.is/files/json/{0}/{1}/",
+    # "Síminn": "https://api.tv.siminn.is/oreo-api/v2/channels/{0}/events?start={1}&end={2}",
 }
 
+# Schedule cache (keep for one day)
+_SCHED_CACHE: cachetools.TTLCache = cachetools.TTLCache(maxsize=15, ttl=86400)
+
+# Type for schedules
+_SchedType = List[Dict[str, Any]]
+
+
+class _AnswerDict(TypedDict):
+    """
+    Format of answer dictionary.
+    Includes answer from module along with station,
+    channel and expiration time.
+    """
+
+    response: Dict[str, Any]
+    answer: str
+    voice: str
+    station: str
+    channel: str
+    expire_time: datetime.datetime
+
+
+# Programs which don't have/need a description
+_NO_DESCRIPTION_SET = frozenset(("fréttir", "fréttayfirlit", "veðurfréttir", "hádegisfréttir"))
+
+
+def _extract_ruv_schedule(response: dict) -> _SchedType:
+    """Safely extract schedule from RÚV API response."""
+    if "error" in response.get("schedule", ""):
+        return []
+    try:
+        return cast(_SchedType, response["schedule"]["services"][0]["events"])
+    except (KeyError, IndexError):
+        return []
+
+
+def _query_schedule_api(channel: str, station: str, date: datetime.date) -> _SchedType:
+    """Fetch and return channel schedule from API or cache for specified date."""
+
+    if (channel, date) in _SCHED_CACHE:
+        return cast(_SchedType, _SCHED_CACHE[(channel, date)])
+
+    if station == "Síminn":
+        # TODO: Síminn endpoint needs its own formatting
+        # since url includes start and end time along with channel ID
+        return []
+    else:
+        url: str = _STATION_ENDPOINTS[station].format(channel, date.isoformat())
+    response = cast(Optional[List], query_json_api(url))
+
+    if response is None:
+        return []
+
+    sched: _SchedType
+    if station == "RÚV":
```

Comment:
Might want to define canonical station names (e.g. "RÚV" string) as constants instead of doing comparisons repeatedly w. bare string.

Reply:
Added constants for frequently used strings.